### PR TITLE
Add support to walk over SQLite changeset and read it as EC instances

### PR DESCRIPTION
**Pull request by khanaffan**
1. `SqliteChangesetReader` - low level api to read changeset from file or local changes from imodel.
2. `ChangesetECAdaptor` - convert SQLite change to EC change thought its done per table and so change is partial.
3. `PartialECChangeUnifier` - combine partial EC change instances into one. This operation need to read whole changeset and require more memory depending on size of changeset.

This API require a context Db for schema and it should belong to same timeline as changeset. It should be as recent as possible.

imodel-native: https://github.com/iTwin/imodel-native/pull/539

### CODE REVIEW

File: common/api/core-backend.api.md [hunk 1]
```diff
@@ -617,6 +620,18 @@ export class CategorySelector extends DefinitionElement {
     toJSON(): CategorySelectorProps;
 }
 
+// @beta (undocumented)
```
Comment (pmconne) [inline common/api/core-backend.api.md:623]: Beta APIs need documentation.

File: core/backend/src/ChangesetECAdaptor.ts [hunk 2]
```diff
@@ -0,0 +1,778 @@
...
+        this._cachedClassMaps.set(classId, classMap);
+        return classMap;
+      }
+      return undefined;
+    });
+  }
+}
+
+/** @beta */
+export interface MetaData {
```
Comment (pmconne) [inline core/backend/src/ChangesetECAdaptor.ts:342]: Need a more specific name than "MetaData".

Comment (khanaffan) [reply core/backend/src/ChangesetECAdaptor.ts:342]: I did not intent to export these classes. They are internal i will remove export from it.

File: core/backend/src/SqliteChangesetReader.ts [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+  readonly db?: AnyDb; /** db from which schema will be read. It should be close to changeset.*/
+  readonly invert?: true; /** invert the changeset operations */
+  readonly disableSchemaCheck?: true; /** do not check if column of change match db schema instead ignore addition columns */
+}
+
+/**
+ * @beta
+ */
+export interface SqliteChange {
+  $table?: string;
```
Comment (pmconne) [inline core/backend/src/SqliteChangesetReader.ts:56]: What motivates prepending `$` to these property names?

Comment (pmconne) [reply core/backend/src/SqliteChangesetReader.ts:56]: Do you expect `[key: string]: any` to contain `op`, `table`, and/or `stage`?

File: core/backend/src/ChangesetECAdaptor.ts (hunk 2 above)
Comment (khanaffan) [reply core/backend/src/ChangesetECAdaptor.ts:342]: Ah wait this is something i need to export I will rename it.

**Review by pmconne — CHANGES_REQUESTED** [review]
Please declare function return types explicitly.

File: core/backend/src/SqliteChangesetReader.ts (hunk 3 above)
Comment (khanaffan) [reply core/backend/src/SqliteChangesetReader.ts:56]: to avoid conflict with column names.

File: common/api/core-backend.api.md (hunk 1 above)
Comment (khanaffan) [reply common/api/core-backend.api.md]: done

File: core/backend/src/ChangesetECAdaptor.ts (hunk 2 above)
Comment (khanaffan) [reply core/backend/src/ChangesetECAdaptor.ts]: renamed

**Comment by pmconne** [comment]
Your corresponding imodel-native PR already merged. Please figure out (perhaps with @DanRod1999's assistance) how to get this PR validated - all the CI jobs appear to have been skipped.

**Review by pmconne — APPROVED** [review]

**Comment by DanRod1999** [comment]
https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=2502906&view=results
This build should validate that this current branch builds with imodel-native main. So if it it is necessary to validate again please wait for this build to succeed.

 I recently changed how our imodel-native pipeline works, and an unintended byproduct was now we have the ability to run the imodel-native pipeline against any desired itwinjs-core branch. However, this is the first real opportunity I've had to test this since the change was made, so I will also keep an eye on it as well.

**Pull request merged**